src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/ServiceController.php

```diff
@@ -281,7 +281,39 @@ public function selectAction(Request $request)
 
         $this->commandBus->handle($command);
 
-        return $this->redirectToRoute('entity_list', ['serviceId' => $serviceId]);
+        if (!$serviceId) {
+            return $this->redirectToRoute('service_overview');
+        }
+
+        return $this->redirectToRoute('service_admin_overview', ['serviceId' => $serviceId]);
+    }
+
+    /**
+     * @Method({"GET"})
+     * @Route("/service/{serviceId}", name="service_admin_overview")
+     * @Security("has_role('ROLE_ADMINISTRATOR')")
+     * @Template("@Dashboard/Service/overview.html.twig")
+     */
+    public function adminOverviewAction($serviceId)
+    {
+        $allowedServices = $this->authorizationService->getAllowedServiceNamesById();
```

Comment:
Just a suggestion: you could use the `\Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Service\AuthorizationService::getServiceById` and the `\Surfnet\ServiceProviderDashboard\Application\Service\ServiceService::getServiceById`. To keep the responsibility for fetching and validating services at the designated service. This will also keep your code flexible and easy to refactor later on. And less lines of code are less prone to errors. :wink:

Reply:
You ara absolutely right, I was in a tunnel vision to get things done. I've greatly simplified the `adminOverviewAction` method.